src/mcp/client/streamable_http.py

```diff
@@ -255,40 +255,50 @@ async def _handle_post_request(self, ctx: RequestContext) -> None:
         headers = self._prepare_request_headers(ctx.headers)
         message = ctx.session_message.message
         is_initialization = self._is_initialization_request(message)
+        try:
+            async with ctx.client.stream(
+                "POST",
+                self.url,
+                json=message.model_dump(by_alias=True, mode="json", exclude_none=True),
+                headers=headers,
+            ) as response:
+                if response.status_code == 202:
+                    logger.debug("Received 202 Accepted")
+                    return
 
-        async with ctx.client.stream(
-            "POST",
-            self.url,
-            json=message.model_dump(by_alias=True, mode="json", exclude_none=True),
-            headers=headers,
-        ) as response:
-            if response.status_code == 202:
-                logger.debug("Received 202 Accepted")
-                return
-
-            if response.status_code == 404:
-                if isinstance(message.root, JSONRPCRequest):
-                    await self._send_session_terminated_error(
-                        ctx.read_stream_writer,
-                        message.root.id,
-                    )
-                return
+                if response.status_code == 404:
```

Comment:
we could instead expand this or add a case like

```
if response.status_code in (500, 502, 503, ...):
```

That does specific handling of only those specific errors you're encountering. That would prevent us from masking other errors that might occur.